csclient/csclient_test.go

```diff
@@ -1196,3 +1189,40 @@ func (s *suite) TestMacaroonAuthorization(c *gc.C) {
 	c.Assert(result.IdRevision.Revision, gc.Equals, rurl.URL.Revision)
 	c.Assert(httpbakery.IsInteractionError(errgo.Cause(err)), gc.Equals, true)
 }
+
+func (s *suite) TestLogin(c *gc.C) {
+	rurl := newResolvedURL("~charmers/utopic/wordpress-42", 42)
+	err := s.store.AddCharmWithArchive(rurl, storetesting.Charms.CharmDir("wordpress"))
+	c.Assert(err, gc.IsNil)
+	err = s.client.Put("/"+rurl.URL.Path()+"/meta/perm/read", []string{"bob"})
```

Comment:
I just copied the code from the previous test, but you're right, they could both benefit from a comment.